[BOLT][AArch64] Fix error message for failed ADR relaxation

maksfb opened the pull request:
Do not recommend the strict mode to the user when ADR relaxation fails on a non-simple function, i.e. a function  with unknown CFG.

We cannot rely on relocations to reconstruct compiler-generated jump tables for AArch64, hence strict mode does not work as intended.

maksfb committed:
[BOLT][AArch64] Fix error message for failed ADR relaxation

Do not recommend the strict mode to the user when ADR relaxation fails
on a non-simple function, i.e. a function  with unknown CFG.

We cannot rely on relocations to reconstruct compiler-generated jump
tables for AArch64, hence strict mode does not work as intended.

maksfb commented on bolt/test/AArch64/r_aarch64_prelxx.s:
```diff
@@ -21,7 +21,7 @@
 // 4 is offset in datatable
 // 8 is addend
-// CHECKPREL32: [[#DATATABLEADDR + 4 - 8 + VALUE]] <_start>:
+// CHECKPREL32: [[#DATATABLEADDR + VALUE]] <_start>:
```
@yavtuk, I had to modify this test case as I believe it was broken. The relocations that are generated for `datatable` are pointing to phantom entry points in `_start`. As ADR in `_start` was previously relaxed into ADRP+ADD, the entry points were shifted by 4 bytes.

maksfb committed:
fixup! [BOLT][AArch64] Fix error message for failed ADR relaxation

paschalis-mpeis reviewed (commented):
Hey Maksim, thanks for the patch.

With this PR we'll also have to adapt: [runtime/AArch64/adrrelaxationpass.s](https://github.com/llvm/llvm-project/blob/abec50cb9391522d7a7019d0cb040e946c8af164/bolt/test/runtime/AArch64/adrrelaxationpass.s#L13)

maksfb committed:
fixup! fixup! [BOLT][AArch64] Fix error message for failed ADR relaxation

maksfb committed:
fixup! fixup! fixup! [BOLT][AArch64] Fix error message for failed ADR relaxation

maksfb commented:
> With this PR we'll also have to adapt: [runtime/AArch64/adrrelaxationpass.s](https://github.com/llvm/llvm-project/blob/abec50cb9391522d7a7019d0cb040e946c8af164/bolt/test/runtime/AArch64/adrrelaxationpass.s#L13)

Thanks for spotting. I've deleted this test and moved useful bits into `adr-relaxation.s`.

maksfb commented on bolt/test/AArch64/adr-relaxation-fail.s:
```diff
@@ -0,0 +1,35 @@
+  b.hi  .L2
+  mov  x0, #0x0
+.L2:
+  adr x1, foo
```
Not really. I removed everything but return.

paschalis-mpeis reviewed (approved):
Hey Maksim,

Great, thanks for the updates, looks good!

yota9 reviewed (approved):
LG thanks
